src/systems/detachable_joint/DetachableJoint.cc

```diff
@@ -234,16 +234,29 @@ void DetachableJoint::PreUpdate(
     }
     else
     {
-      auto candidateEntities = entitiesFromScopedName(
+      auto entitiesMatchingName = entitiesFromScopedName(
         this->childModelName, _ecm);
 
+      // TODO(arjoc): There is probably a more efficient way of combining entitiesFromScopedName
+      // With filtering.
+      // Filter for entities with only models
+      std::vector<Entity> candidateEntities;
+      for (const auto entity: entitiesMatchingName)
+      {
+        if (_ecm.EntityHasComponentType(entity, components::Model::typeId))
+        {
+          candidateEntities.push_back(entity);
+        }
+      }
```

Comment:
```suggestion
      std::copy_if(entitiesMatchingName.begin(), entitiesMatchingName.end(),
                 std::back_inserter(candidateEntities),
                 [](Entity e) { return _ecm.EntityHasComponentType(e, components::Model::typeId); });
```

Reply:
Sorry, I made this suggestion against an older revision. Will make a fresh suggestion on latest revision.